### DISCUSSION TITLE
Ory cookies read by all subdomains

### DISCUSSION THREAD

**Discussion by aleti123** [ory/kratos#3625] [Q&A] [answered]
I have created an application and hosted on dev & production. This how the urls looks like `app.hello.com`, `app.dev.hello.com` (adding random urls). 

I write a go server which does authentication for APIs. This how the code looks like

```
// GetSession get the user session.
func (o *Config) GetSession(context context.Context, cookie string) (*oryClient.Session, error) {
	session, res, err := o.client.FrontendApi.ToSession(context).Cookie(cookie).Execute()
	if res == nil {
		return nil, errors.New("unable to fetch anything from ory")
	}
	if res.Body != nil {
		defer res.Body.Close()
	}
	if res.StatusCode == 401 {
		return nil, ErrUnAuthorized
	}
	if err != nil {
		return nil, err
	}
	if !*session.Active {
		return nil, ErrUnAuthorized
	}
	return session, nil
}

cookies is `r.Cookie`
```

In frontend, I am using `Axios`, and set `withCredentials:true` to read the cookies and send it in request headers.

* When I open `app.hello.com` only cookies with `hello.com` is sent to server. There is not issue here.
* When I open `app.dev.hello.com` then all the cookies has either `hello.com`, or `dev.hello.com` sent to server. I need to restrict this to only `dev.hello.com`. And above function `GetSession` throwing error.

Can someone help me with this?

**Comment by mklebrasseur** [comment, marked as answer]
If you are running 2 different Kratos instances you should simply be able to set the domain for both

On Production
```
session:
  cookie:
    domain: hello.com
```

On Development
```
session:
  cookie:
    domain: dev.hello.com
```
